### PR TITLE
Added ClickTp

### PR DESCRIPTION
**Pull request by o0F-0oF**
ClickTp

### CODE REVIEW

File: KiraiMod/Modules/ClickTp.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            };
+            GUI.Groups.Loaded += () =>
+            {
+                UIGroup ClickTp = new("ClickTp", GUI.Groups.Movement);
+                ClickTp.AddElement("ClickTp", clickTp.Value).Bound.Bind(clickTp);
+            };
+            clickTp.SettingChanged += ((EventHandler)((sender, args) =>
+            {
+                Events.Update += () => RaycastTp();
+            })).Invoke();
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: the method is never removed from Update when the setting is disabled

File: KiraiMod/Modules/ClickTp.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using KiraiMod.Core.UI;
+using System;
+using UnityEngine;
+using VRC.SDKBase;
+
+namespace KiraiMod.Modules
+{
+    public static class ClickTp
+    {
+        public static ConfigEntry<bool> clickTp = Plugin.cfg.Bind("ClickTp", "Toggle", false, "Should you be able to Tp using left ctrl and left mouse click");
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: this should be called "enabled"

File: KiraiMod/Modules/ClickTp.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿using BepInEx.Configuration;
+using KiraiMod.Core.UI;
+using System;
+using UnityEngine;
+using VRC.SDKBase;
+
+namespace KiraiMod.Modules
+{
+    public static class ClickTp
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: this is an inelegant class name, i would change it to `ClickTeleport`

File: KiraiMod/Modules/ClickTp.cs (hunk 2 above)
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: change the section the same was the class name should be

File: KiraiMod/Modules/ClickTp.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        public static ConfigEntry<bool> clickTp = Plugin.cfg.Bind("ClickTp", "Toggle", false, "Should you be able to Tp using left ctrl and left mouse click");
+        public static ConfigEntry<int> tpRange = Plugin.cfg.Bind("ClickTp", "Tp Range", 100000, "How far should you be able to Tp");
+        public static Camera cameraTarget;
+        static ClickTp()
+        {
+            Events.UIManagerLoaded += () =>
+            {
+                cameraTarget = Camera.main;
+            };
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: braces are unneeded

File: KiraiMod/Modules/ClickTp.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            clickTp.SettingChanged += ((EventHandler)((sender, args) =>
+            {
+                Events.Update += () => RaycastTp();
+            })).Invoke();
+        }
+        public static void RaycastTp()
+        {
+            RaycastHit hit;
+            if (!clickTp.Value) return;
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: this check is unneeded if you remove the method from Update correctly

File: KiraiMod/Modules/ClickTp.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            if (Input.GetKey(KeyCode.LeftControl) && Input.GetKeyUp(KeyCode.Mouse0))
```
Comment (xKiraiChan) [inline KiraiMod/Modules/ClickTp.cs]: this does not support VR

**Review by xKiraiChan — CHANGES_REQUESTED** [review]
use formal names like `ClickTeleport` instead of just `Tp`

File: KiraiMod/Modules/ClickTp.cs (hunk 6 above)
Comment (o0F-0oF) [reply KiraiMod/Modules/ClickTp.cs]: fuck vr support

**Comment by xKiraiChan** [comment]
in the future this needs VR support

**Comment by o0F-0oF** [comment]
ok

**Pull request merged**